packages/experiments/src/components/Toggle/Toggle.state.ts

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { IToggleProps, IToggleViewProps } from './Toggle.types';
+import { BaseState } from '../../utilities/BaseState';
+
+export type IToggleState = Pick<IToggleViewProps, 'checked' | 'onChange' | 'onClick' | 'text'>;
+
+export class ToggleState extends BaseState<IToggleProps, IToggleViewProps, IToggleState> {
+  constructor(props: ToggleState['props']) {
+    super(props);
+
+    this.state = {
+      checked: !!props.defaultChecked,
+      text: !!props.defaultChecked ? props.onText : props.offText,
+      onChange: this._noop,
+      onClick: this._onClick
+    };
+  }
+
+  private _onClick = (ev: React.MouseEvent<HTMLElement>) => {
+    const { checked: checkedProp, onText, offText, disabled, onChange } = this.props;
+    const { checked } = this.state;
+
+    if (!disabled) {
+      // Only update the state if the user hasn't provided it.
+      if (checkedProp === undefined) {
```

Comment:
Actually you don't have to do this explicit check. The `BaseState` constructor takes a second argument which is a list of controlled values. In this case, it should be `{ controlledProps: ['checked'] }`. [Button](https://github.com/OfficeDev/office-ui-fabric-react/blob/0661c73b152466194d9715477b1f86496eb4eeea/packages/experiments/src/components/Button/Button.state.ts#L10) does something similar. This allows you to set `checked` via `setState` without fear because `BaseState` will ensure the proper value is passed to the view.

Reply:
Although it does raise an interesting question with `text`.. what if the state's value disagrees with the prop? Then we'll show the wrong text? Hmmm...

Reply:
It is an interesting question for sure. Should we leave this check? Should we leave the check but only for `text` and let `BaseState` handle `checked`?

I just went ahead and tried deleting the condition and the text was indeed changing although the pill wasn't being toggled.

Reply:
Another thing we could do is retrieve the state of `checked` once again after calling `this.setState()` and use that value to set `text`. Let me know which is the option you prefer.